library/src/main/java/pl/rspective/voucherify/android/client/api/VoucherifyApi.java

```diff
@@ -1,12 +1,13 @@
 package pl.rspective.voucherify.android.client.api;
 
+import pl.rspective.voucherify.android.client.exception.VoucherifyError;
 import pl.rspective.voucherify.android.client.model.VoucherResponse;
 import retrofit.http.GET;
 import retrofit.http.Query;
 
 public interface VoucherifyApi {
 
     @GET("/client/v1/validate")
-    VoucherResponse validateVoucher(@Query("code") String code, @Query("tracking_id") String trackingId, @Query("channel") String channel);
+    VoucherResponse validateVoucher(@Query("code") String code, @Query("tracking_id") String trackingId, @Query("channel") String channel) throws VoucherifyError;
```

Comment:
Yes, it is proper and even required. Javadoc to retrofit's [`ErrorHandler.handleError()`](https://square.github.io/retrofit/1.x/retrofit/retrofit/ErrorHandler.html#handleError-retrofit.RetrofitError-) says:
> If the return exception is checked it must be declared to be thrown on the interface method.

Declaring `VoucherifyError` as a checked exception helped me to track all the way down where to handle it. Since it's finished now I can turn it into RuntimeException.